primitives/externalities/src/lib.rs

```diff
@@ -47,7 +47,7 @@ pub trait Externalities: ExtensionStore {
 	/// Returns an `Option` that holds the SCALE encoded hash.
 	fn child_storage_hash(
 		&self,
-		storage_key: ChildStorageKey,
+		storage_key: &[u8],
```

Comment:
Is there a reason why you're using `&[u8]` here instead of `StorageKey` from `use sp_core::storage::{StorageKey};`? Can the `StorageKey` be of any/ variable length here?

Reply:
Yest storage can can be of any/variable length, using a slice here is possible as the storage key does not change, I could have use '&ChildStorageKey' instead, but that is consistent with the way `key` parameter is used.

Note that I am in the process of doing another pass of refactoring, simplifying a lot more, I hope it will be finished this evening :)